cpp/src/arrow/compute/kernels/isin-test.cc

```diff
@@ -0,0 +1,410 @@
+                               {false, false, false, false}, {});
+
+  // Nulls in right array
+  CheckIsIn<BooleanType, bool>(&this->ctx_, boolean(), {true, true, false, true}, {},
+                               {true, true, false, true}, {false, false, false, false},
+                               {false, false, false, false}, {});
+
+  // Both array have Nulls
+  CheckIsIn<BooleanType, bool>(&this->ctx_, boolean(), {false, true, false, true},
+                               {false, false, false, false}, {true, true, true, true},
+                               {false, false, false, false}, {true, true, true, true},
+                               {});
+}
+
+TEST_F(TestIsInKernel, IsInBinary) {
+  CheckIsIn<BinaryType, std::string>(
+      &this->ctx_, binary(), {"test", "", "test2", "test"}, {true, false, true, true},
+      {"test", "", "test2", "test"}, {true, false, true, true}, {true, true, true, true},
+      {});
+
+  // No match
+  CheckIsIn<BinaryType, std::string>(
+      &this->ctx_, binary(), {"test", "", "test2", "test"}, {true, false, true, true},
+      {"test3", "test4", "test3"}, {true, true, true}, {false, false, false, false}, {});
+
+  // Nulls in left array
+  CheckIsIn<BinaryType, std::string>(
+      &this->ctx_, binary(), {"test", "", "test2", "test"}, {false, false, false, false},
+      {"test", "test2", "test"}, {true, true, true}, {false, false, false, false}, {});
+
+  // Nulls in right array
+  CheckIsIn<BinaryType, std::string>(&this->ctx_, binary(), {"test", "test2", "test"},
+                                     {true, true, true}, {"test", "", "test2", "test"},
+                                     {false, false, false, false}, {false, false, false},
+                                     {});
+
+  // Both array have Nulls
+  CheckIsIn<BinaryType, std::string>(
+      &this->ctx_, binary(), {"test", "", "test2", "test"}, {false, false, false, false},
+      {"test", "", "test2", "test"}, {false, false, false, false},
+      {true, true, true, true}, {});
+
+  // Empty arrays
+  CheckIsIn<BinaryType, std::string>(&this->ctx_, binary(), {}, {}, {}, {}, {}, {});
+
+  // Empty left array
+  CheckIsIn<BinaryType, std::string>(&this->ctx_, binary(), {}, {},
+                                     {"test", "", "test2", "test"},
+                                     {true, false, true, false}, {}, {});
+
+  // Empty right array
+  CheckIsIn<BinaryType, std::string>(&this->ctx_, binary(), {"test", "", "test2", "test"},
+                                     {true, false, true, true}, {}, {},
+                                     {false, false, false, false}, {});
+
+  CheckIsIn<StringType, std::string>(
+      &this->ctx_, utf8(), {"test", "", "test2", "test"}, {true, false, true, true},
+      {"test", "", "test2", "test"}, {true, false, true, false}, {true, true, true, true},
+      {});
+
+  // Nulls in left array
+  CheckIsIn<StringType, std::string>(
```

Comment:
can we consolidate these with the binary tests in some way?

Reply:
Changed it to a Typed_Test for BinaryType and StringType.